ddtrace/contrib/django/utils.py

```diff
@@ -88,6 +87,15 @@ def get_request_uri(request):
 
     # Build request url from the information available
     # DEV: We are explicitly omitting query strings since they may contain sensitive information
-    return parse.urlunparse(
-        parse.ParseResult(scheme=request.scheme, netloc=host, path=request.path, params="", query="", fragment="",)
-    )
+    urlparts = dict(scheme=request.scheme, netloc=host, path=request.path, params=None, query=None, fragment=None)
+
+    # DEV: In PY3, urlunparse calls urllib.parse._coerce_args which uses the
+    # type of the scheme to check the type to expect from all url parts, raising
+    # a TypeError otherwise
+    # https://github.com/python/cpython/blob/02d126aa09d96d03dcf9c5b51c858ce5ef386601/Lib/urllib/parse.py#L111-L125
+    # Unless all url parts are bytes, we must then cast to strings to avoid the
+    # TypeError
+    if PY3 and not all(isinstance(value, binary_type) or value is None for value in urlparts.values()):
```

Comment:
We only want to convert to string in all cases other than when all values are bytes, so we shouldn't remove that conditional. Is your suggestion to move away from list comprehensions?

Reply:
I did include that logic in the suggested change below with the tweak to the comprehension condition (I think), but yes I think this would be quite a bit cleaner as a plain ol for loop

Reply:
Since we need to first check if all parts are bytes, we'd loop twice. I think the first which is used to determine if we should cast is best done as a list comprehension with `all` as done here. The update could be rolled out into a loop since it has got some conditionals.

Reply:
Ohhh kay think I got it. Strange behaviour! I should have read the comment above more thoroughly.

If I've got this right: If all the values are bytes we're ok, if not then we want to cast them all to a string type.

Is the bytes check really necessary i.e. if we end up `to_unicode`ing them all would it be fine? I think this is what I was going for with my suggestions as it seems that they end up getting encoded/cast to string anyway by _coerce_args.

Reply:
@Kyle-Verhoog my understanding of `_coerce_args` is that, yes, `_decode_args` will decode bytes to string, but it also returns the `_encode_result` function if the scheme is not a `str` (and if no TypeError is raised). Then later `_encode_result` will transform the string based on the encoding used to decode the args. If we didn't do the bytes check and `to_unicode` all, our helper would make never return a byte string.